Marlin/planner.cpp

```diff
@@ -1206,7 +1206,7 @@ void Planner::_buffer_line(const float &a, const float &b, const float &c, const
     // This leads to an enormous number of advance steps due to a huge e_acceleration.
     // The math is correct, but you don't want a retract move done with advance!
     // So this situation is filtered out here.
-    if (!block->steps[E_AXIS] || (!block->steps[X_AXIS] && !block->steps[Y_AXIS] && !block->steps[Z_AXIS]) || stepper.get_advance_k() == 0 || (uint32_t) block->steps[E_AXIS] == block->step_event_count) {
+    if (!block->steps[E_AXIS] || (!block->steps[X_AXIS] && !block->steps[Y_AXIS]) || stepper.get_advance_k() == 0 || (uint32_t) block->steps[E_AXIS] == block->step_event_count) {
```

Comment:
This solves the bug from #4896, see [posts around here.](https://github.com/MarlinFirmware/Marlin/issues/4896#issuecomment-253256752)

In a nutshell, we have to prevent LIN_ADVANCE to be active during Z & E only moves because that's a combination where advancing the stepper makes no sense. If not corrected, it will lead to blobs with retract/prime combined with Z-hops.